setup.iss

```diff
@@ -0,0 +1,242 @@
+Source: "RetroBar\bin\Release\{#TargetFramework}\publish-x86\*"; DestDir: "{app}"; Check: PreferX86Files; Flags: solidbreak ignoreversion recursesubdirs
+
+[UninstallDelete]
+Type: filesandordirs; Name: "{localappdata}\RetroBar\Logs"
+Type: files; Name: "{localappdata}\RetroBar\settings.json"
+Type: dirifempty; Name: "{localappdata}\RetroBar"
+
+[Icons]
+Name: "{autoprograms}\{#RetroBarName}"; Filename: "{app}\{#RetroBarExeName}"
+Name: "{autodesktop}\{#RetroBarName}"; Filename: "{app}\{#RetroBarExeName}"; Tasks: desktopicon
+
+[Registry]
+Root: HKA; Subkey: "Software\Microsoft\Windows\CurrentVersion\Run"; ValueName: "RetroBar"; ValueType: string; ValueData: "{app}\{#RetroBarExeName}"; Tasks: autostart; Flags: uninsdeletevalue
+
+[Run]
+Filename: "{app}\{#RetroBarExeName}"; Description: "{cm:LaunchProgram,{#StringChange(RetroBarName, '&', '&&')}}"; Flags: nowait postinstall skipifsilent
+Filename: "{tmp}\{#DotNetInstallerExe}"; StatusMsg:"Installing Microsoft .NET 6 Desktop Runtime..."; Parameters:"/install /norestart"; Description: "{cm:LaunchProgram,{#StringChange(RetroBarName, '&', '&&')}}"; Check: NetRuntimeIsMissing; Flags: skipifdoesntexist
+
+[Code]
+var
+  DownloadPage: TDownloadWizardPage;
+  DotNetChecked: boolean;
+  DotNetMissing: boolean;
+ 
+function PreferArm64Files: Boolean;
+begin
+  Result := IsArm64;
+end;
+
+function PreferX64Files: Boolean;
+begin
+  Result := not PreferArm64Files and IsX64Compatible;
+end;
+
+function PreferX86Files: Boolean;
+begin
+  Result := not PreferArm64Files and not PreferX64Files;
+end;
+
+// Version check courtesy https://engy.us/blog/2021/02/28/installing-net-5-runtime-automatically-with-inno-setup/
+function CompareVersion(V1, V2: string): Integer;
+var
+  P, N1, N2: Integer;
+begin
+  Result := 0;
+  while (Result = 0) and ((V1 <> '') or (V2 <> '')) do
+  begin
+    P := Pos('.', V1);
+    if P > 0 then
+    begin
+      N1 := StrToInt(Copy(V1, 1, P - 1));
+      Delete(V1, 1, P);
+    end
+      else
+    if V1 <> '' then
+    begin
+      N1 := StrToInt(V1);
+      V1 := '';
+    end
+      else
+    begin
+      N1 := 0;
+    end;
+    P := Pos('.', V2);
+    if P > 0 then
+    begin
+      N2 := StrToInt(Copy(V2, 1, P - 1));
+      Delete(V2, 1, P);
+    end
+      else
+    if V2 <> '' then
+    begin
+      N2 := StrToInt(V2);
+      V2 := '';
+    end
+      else
+    begin
+      N2 := 0;
+    end;
+    if N1 < N2 then Result := -1
+      else
+    if N1 > N2 then Result := 1;
+  end;
+end;
+
+function NetRuntimeIsMissing(): Boolean;
+var
+  runtimes: TArrayOfString;
+  registryKey: string;
+  I: Integer;
+  meetsMinimumVersion: Boolean;
+  meetsMaximumVersion: Boolean;
+  minimumVersion: string;
+  maximumExclusiveVersion: string;
+begin
+  Result := True;
+  
+  if DotNetChecked then
+  begin
+    Result := DotNetMissing;
+    Exit;
+  end;
+
+  DotNetChecked := True;
+  DotNetMissing := True;
+  minimumVersion := '6.0.0';
```

Comment:
It should use the version used at compile time, this probably won't work because current RetroBar builds can't run until .NET 6.0 is at least updated to the version used for compilation. This is really strict.

Reply:
Oh, it should work with 6.0.2 or later.